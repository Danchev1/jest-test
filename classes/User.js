export default class User {
  constructor(details) {
    let { firstname, lastname } = details;
    this.firstname = firstname;
    this.lastname = lastname;
  }

  get name() {
    return `${this.firstname} ${this.lastname}`;
  }

  userDescription() {
    return `This is a short description of the user: ${this.name}`
  }

}
