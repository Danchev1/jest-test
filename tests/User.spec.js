import User from '../classes/User';

describe('User.js', () => {
  test('Should return full name', () => {
    let user = new User({
      firstname: 'Shoolboy',
      lastname: 'Q'
    });

    expect(user.name).toBe('Shoolboy Q');
  });

  test('Should match user description', () => {
    let userDesc = new User({
      firstname: 'John',
      lastname: 'Whick'
    }).userDescription();

    expect(userDesc).toMatchSnapshot();
  });
});
