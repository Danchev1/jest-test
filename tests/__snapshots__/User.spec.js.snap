// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User.js Should match user description 1`] = `"This is a short description of the user: John Whick"`;
